### ISSUE TITLE
Should flags be represented as enums?

### ISSUE THREAD

**Issue by xtian** [marlersoft/zigwin32gen#4]
Regarding this TODO: https://github.com/marlersoft/zigwin32gen/blob/b42acec570a35ece109d3d8cd740820171147ad3/src/genzig.zig#L1669

Is there a big benefit to representing these values as non-exhaustive enums? My impression is that the common case is combining multiple bitmask values together which is awkward with the current setup. For example:

```zig
const std = @import("std");
const win32 = @import("win32").everything;

const class = std.mem.zeroInit(win32.WNDCLASSEXA, .{
    .style = @intToEnum(win32.WNDCLASS_STYLES, @enumToInt(win32.CS_HREDRAW) | @enumToInt(win32.CS_VREDRAW) | @enumToInt(win32.CS_OWNDC)),
});
```

**Comment by marler8997** [comment]
Yeah I agree that in the case of enum flags, you always have to convert to an int and back to the enum which throws type safety out the window.  I'd be curious if anyone has a better idea than just raw integer types though.

**Comment by marler8997** [comment]
Looks like this zig proposal could be a solution: https://github.com/ziglang/zig/issues/5049

It updates packed structs to specify and underlying integer type so they are compatible with the C abi.  If that proposal is going to be accepted, we might be able to start transitioning to it beforehand.

**Comment by marler8997** [comment]
Ok I came up with one idea, I've added an `initFlags` method to all enum flag types for now.  So your example could become this:
```zig
const class = std.mem.zeroInit(win32.WNDCLASSEXA, .{
    .style = win32.WNDCLASS_STYLES.initFlags(.{.HREDRAW = 1, .VREDRAW = 1, .OWNDC = 1});
});
```

Full API diff for this change is here: https://github.com/marlersoft/zigwin32/commit/89203c6156a0d2206f61ae75db69bd9fbbdd1adc#diff-bd4f828bddb85d1324457f27ad66278624ef2539c5036de4c596d3b15a695820R3458

It affected 11,500 lines!

**Comment by xtian** [comment]
@marler8997 Ok that seems like a decent solution for now. I like the idea of integer-backed packed structs though.

Here's at least one case that your solution doesn't handle:

`VIRTUAL_ALLOCATION_TYPE` is evidently not marked as "Flags", but [`VirtualAlloc` can be called with `MEM_COMMIT | MEM_RESERVE`](https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc#:~:text=are%20actually%20accessed.-,To%20reserve%20and%20commit%20pages%20in%20one%20step%2C%20call%20VirtualAlloc%20with%20MEM_COMMIT%20%7C%20MEM_RESERVE.,-Attempting%20to%20commit).

**Comment by marler8997** [comment]
@xtian good catch. I've created an issue with the win32metadata project here: https://github.com/microsoft/win32metadata/issues/507

**Comment by marler8997** [comment]
Just did a new zigwin32 release.  Looks like `VIRTUAL_ALLOCATION_TYPE` is not properly marked with the `[Flags]` attribute (see https://github.com/marlersoft/zigwin32/blob/146598875015d7b1597af579ce0d41a93e1f4144/win32/system/memory.zig#L374).

**Comment by capthehacker99** [comment]
So how do you check if a flag exist?
Mentioned in this issue: https://github.com/marlersoft/zigwin32/issues/28

**Comment by marler8997** [comment]
zigwin32 has moved on to using packed structs to represent flags which are very nice to use, so this issue should be addressed now.
